tests/pykafka/test_producer.py

```diff
@@ -118,6 +126,30 @@ def test_null_payloads(self):
         prod.produce(b"")  # empty string should be distinguished from None
         self.assertEqual(b"", self.consumer.consume().value)
 
+    def test_cleanup_stop_is_not_called_on_stopped_object(self):
+        """Test that the stop() method is not called twice"""
+        prod = self.client.topics[self.topic_name].get_producer()
+        prod.stop()
+        self.assertFalse(hasattr(prod, '_cleanup_func'))
+
+    def test_cleanup_stop_is_called_on_not_stopped_object(self):
+        """Test that the cleanup() method flushes the queue properly"""
```

Comment:
I really like this docstring putting it in these terms. Even though this is implicitly also tested in `test_async_produce_context`, it is actually a requirement that deserves the emphasis given here.